OneSignalSDK/onesignal/src/main/java/com/onesignal/OSUtils.java

```diff
@@ -170,6 +170,21 @@ private boolean hasHMSPushKitLibrary() {
       }
    }
 
+   private static boolean hasHMSAGConnectLibrary() {
+      try {
+         // noinspection ConstantConditions
+         return com.huawei.agconnect.config.AGConnectServicesConfig.class != null;
+      } catch (Throwable e) {
```

Comment:
Should we keep this as `Throwable` or find the specific `Exception(s)`?

Reply:
These will throw a `ClassNotFoundException` at runtime if missing, however if I try to catch it here the compiler thinks it's not possible won't compile.

The best we can do is change these to `Exception`, which is a bit better than `Throwable`